src/katsdpmodels/fetch/aiohttp.py

```diff
@@ -162,3 +178,58 @@ async def fetch_model(url: str, model_class: Type[_M], *,
     model_cache: Dict[str, models.Model] = {}
     async with Fetcher(session=session, model_cache=model_cache) as fetcher:
         return await fetcher.get(url, model_class)
+
+
+class TelescopeStateFetcher(fetch.TelescopeStateFetcherBase):
```

Comment:
Given that the docstring and __init__ are duplicated between the async and normal version (apart from the typing on telstate), shouldn't we have an intermediate class that these two inherit from ?

Reply:
I've moved the docstring to the base class, and assigned to `__doc__` to make it appear in the subclasses.

Re `__init__`, I could parametrise the base class by the fetcher class as well and have the constructor take a factory, but I think by the time I'm done with that and all the annotations so that mypy knows what type the fetcher is in each sub-class, I won't have saved any lines of code, and introduced more complexity. I'd also rather avoid splitting responsibility for managing the fetcher object between base- and sub-class: `close` is responsible for destroying the fetcher, and can't be in the base class because it's a coroutine in one case, so I'd rather have `__init__` of the same class be responsible for creating it.

Reply:
Fair enough.